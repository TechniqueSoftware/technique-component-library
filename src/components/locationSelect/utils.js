export const urlClubIdRegex = id => new RegExp(`^.*\/${id}(\/)?(#)?(((\/)?(#){1}){1}([0-9a-zA-Z])+?)*?$`, 'gm') // eslint-disable-line no-useless-escape, max-len
const urlLocationIdRegex = id => new RegExp(`^.*\/${id}\/`, 'gm') // eslint-disable-line no-useless-escape

export const getLocationIdFromUrl = (path, clubId) => {
  try {
    const [urlBeginning] = path.match(urlLocationIdRegex(clubId))
    const locationId = path.substring(urlBeginning.length).replace(/[^0-9]/g, '')
    return locationId ? parseInt(locationId, 10) : null
  } catch (err) {
    return null
  }
}
