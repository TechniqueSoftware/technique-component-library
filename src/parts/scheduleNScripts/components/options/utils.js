export const followUpTimeToObject = followUpTime => {
  const getHour = h => {
    if (h === '00') return 12
    const parsedHour = parseInt(h, 10)
    return parsedHour > 12 ? parsedHour - 12 : parsedHour
  }
  const time = followUpTime.split(':')
  const [oldHour, oldMinute] = time
  const hour = getHour(oldHour)
  const minute = parseInt(oldMinute, 10)
  const meridiem = oldHour >= 12 ? 'PM' : 'AM'
  return { hour, minute, meridiem }
}

export const objectToFollowUpTime = (hour, minute, meridiem) => {
  const parseHour = h => meridiem === 'PM' ? parseInt(h, 10) + 12 : h
  return `${parseHour(hour === 12 ? '00' : hour)}:${minute === 0 ? '00' : minute}:00`
}
