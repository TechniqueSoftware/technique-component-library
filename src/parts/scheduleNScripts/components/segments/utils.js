import { FOLLOWUP_STATUS } from './constants'

export const parseTitle = followUpStatus => {
  switch (followUpStatus) {
    case FOLLOWUP_STATUS.WEB_LEAD:
      return 'Web Lead'
    case FOLLOWUP_STATUS.APPT_BOOKED:
      return 'Appt Booked'
    case FOLLOWUP_STATUS.MEMBER:
      return 'Member'
  }
}

export const parseLocations = segmentLocations => segmentLocations.map(({ locationName, locationId }) => ({
  label: `${locationName}`,
  value: locationId
}))

export const checkPermission = userPermission => userPermission.some(
  element => element.permissionName === 'CopySegmentsAction' && element.permissionEffect === 'ALLOW')
