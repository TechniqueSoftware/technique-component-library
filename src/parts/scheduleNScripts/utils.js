import { PERMISSION_KEYS, STATUSES_WITH_SEGMENTS } from './constants'

export const checkIfSegmentStatus = (type, id) => {
  const hasSegmentation = !!STATUSES_WITH_SEGMENTS.find(s => s.type === parseInt(type, 10) && s.id === parseInt(id, 10))
  return hasSegmentation
}

const setSchedulesDayNum = schedules => schedules.reduce(
  (acc, curr) => {
    const dayNum = curr.daysTillNext + acc.dayNum
    return { schedules: acc.schedules.concat({ ...curr, dayNum }), dayNum }
  },
  { dayNum: 0, schedules: [] }
).schedules

export const getStatusSegments = (type, id, segments) => {
  if (!checkIfSegmentStatus(type, id)) return []
  const currentStatus = segments.find(
    ({ followUpType, followUpStatus }) => followUpType === parseInt(type, 10) && followUpStatus === parseInt(id, 10)
  )
  if (!currentStatus) return []
  const { followUpSegments = [], followUpSchedules = [], followUpSegmentScripts = [] } = currentStatus
  return followUpSegments.map(segment => {
    const schedulesWithScripts = followUpSchedules.reduce((acc, schedule) => {
      if (segment.segmentId !== schedule.segmentId) return acc
      return acc.concat({
        ...schedule,
        scripts: followUpSegmentScripts.filter(
          script => script.followUpSeqNum === schedule.followUpSeqNum && script.segmentId === schedule.segmentId
        )
      })
    }, [])
    return { ...segment, followUpSchedules: setSchedulesDayNum(schedulesWithScripts) }
  })
}

export const sortSegments = segments => [...segments].sort((a, b) => a.name.localeCompare(b.name))

// Formatting club-os data into something usable
export const parseSnSData = (data, followUpSchedules, followUpScripts, segmentsData, isClub) => {
  const schedulesPermissions = followUpSchedules.permissions
  const scriptsPermissions = followUpScripts.permissions
  const schedules = isClub ? followUpSchedules : followUpSchedules.schedules
  const scripts = isClub ? followUpScripts : followUpScripts.scripts

  const group = (acc, curr) => {
    const { followUpStatus, followUpType } = curr
    const key = `${followUpStatus}-${followUpType}`
    if (acc[key]) acc[key].push(curr)
    else acc[key] = [curr]
    return acc
  }

  const sch = schedules.reduce(group, {})
  const scr = scripts.reduce(group, {})

  Object.keys(sch).forEach(key => { sch[key] = setSchedulesDayNum(sch[key]) })

  Object.keys(scr).forEach(key => {
    const scripts = scr[key]
    scripts.forEach(script => {
      const schedule = sch[key]?.find(el => el.followUpSeqNum === script.followUpSeqNum)
      if (!schedule) return
      if (schedule.scripts) schedule.scripts.push(script)
      else schedule.scripts = [script]
    })
  })

  const followups = data.map(el => {
    const { type, statuses, id } = el
    const canEditSchedules = isClub ? undefined : schedulesPermissions[PERMISSION_KEYS.SCHEDULES[type]]
    const canEditScripts = isClub ? undefined : scriptsPermissions[PERMISSION_KEYS.SCRIPTS[type]]
    const newStatuses = statuses.map(st => {
      const key = `${st.id}-${type}`
      const unsortedSegments = getStatusSegments(type, st.id, segmentsData)
      const segments = sortSegments(unsortedSegments)
      return { ...st, followUpTab: id, schedules: sch[key] || [], segments }
    })
    return { ...el, statuses: newStatuses, canEditSchedules, canEditScripts }
  })

  return {
    followups,
    canEditSchedules: followups[0].canEditSchedules,
    canEditScripts: followups[0].canEditScripts
  }
}

export const getSchedulesByDay = (schedules, dayNum, order) => {
  const { earlierSch, sameDaySch, laterSch } = schedules.reduce(
    (acc, sch) => {
      if (sch.dayNum < dayNum) acc.earlierSch = acc.earlierSch.concat(sch)
      if (sch.dayNum === dayNum) acc.sameDaySch = acc.sameDaySch.concat(sch)
      if (sch.dayNum > dayNum) acc.laterSch = acc.laterSch.concat(sch)
      return acc
    },
    { earlierSch: [], sameDaySch: [], laterSch: [] }
  )
  return {
    earlierSch: order === 'first' ? earlierSch : earlierSch.concat(sameDaySch),
    laterSch: order === 'first' ? sameDaySch.concat(laterSch) : laterSch
  }
}

export const getCurrentSchedules = (segmentId, status) => {
  if (!segmentId) return status.schedules
  return status.segments.find(segment => segment.segmentId === segmentId).followUpSchedules
}

export const getStatus = (followups, type, status) => {
  const fs = followups.filter(f => f.type === type)
  if (!fs.length) return null

  let st = null
  for (let i = 0; i < fs.length; i++) {
    st = fs[i].statuses.find(s => s.id === `${status}`)
    if (st) break
  }
  return st
}

export const setFollowUpStatusValue = (followUps, status, value) => {
  const newFollowups = followUps.map(followUp => {
    if (followUp.id !== status.followUpTab) return followUp
    const newStatuses = followUp.statuses.map(st => {
      if (st.id !== status.id) return st
      return { ...st, ...value }
    })
    return { ...followUp, statuses: newStatuses }
  })
  return newFollowups
}

export const setFollowUpScripts = (followUps, { followUpType, followUpStatus }, scriptsData, segmentId) => {
  const status = getStatus(followUps, followUpType, followUpStatus)
  const currentSchedules = getCurrentSchedules(segmentId, status)
  const schedules = currentSchedules.map(schedule => {
    if (schedule.followUpSeqNum !== scriptsData.followUpSeqNum) return schedule
    return scriptsData
  })
  const statusValueToUpdate = !segmentId
    ? { schedules }
    : {
      segments: status.segments.map(
        segment => (segment.segmentId !== segmentId ? segment : { ...segment, followUpSchedules: schedules }) // eslint-disable-line max-len
      )
    }
  return setFollowUpStatusValue(followUps, status, statusValueToUpdate)
}

export const updateTouchpointInFollowups = (followUps, tp, data) => {
  const { followUpType, followUpStatus, followUpSeqNum } = tp
  const status = getStatus(followUps, followUpType, followUpStatus)
  const newSchedules = status.schedules.map(sch => {
    if (sch.followUpSeqNum !== followUpSeqNum) return sch
    const newSch = { ...sch, ...data }
    return newSch
  })
  return setFollowUpStatusValue(followUps, status, { schedules: newSchedules })
}
