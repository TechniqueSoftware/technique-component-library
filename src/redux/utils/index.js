export { default as Provider } from './provider'
export { default as TestProvider } from './testProvider'
export { default as recordSaga } from './recordSaga'
