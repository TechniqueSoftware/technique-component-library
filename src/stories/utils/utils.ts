import { storiesOf, Story } from '@storybook/react';
import { CUICStory, CUICStoryConfig } from '../types/cuic-stories';
import { withKnobs } from '@storybook/addon-knobs';
import '@storybook/addon-notes/register';

/**
 * A method that gets added path our CUICStoriesOf object. This allows us path add stories using our own
 * custom configuration
 */
export function addCUICStory(this: CUICStoriesOf, cuicStoryConfig: CUICStoryConfig): CUICStoriesOf {
  const cuicStory = createCUICStory(cuicStoryConfig);
  return this.add(cuicStory.name, cuicStory.render, cuicStory.context);
}

type CUICStoriesOf = { addCUICStory: typeof addCUICStory } & Story;

/**
 * Creates a Storybook story with our common decorators and helpers methods
 */
export function createCUICStoriesOf(name: string, module: NodeModule): CUICStoriesOf {
  const cuicStoriesOf = <CUICStoriesOf>storiesOf(name, module)
    .addDecorator(withKnobs);

  cuicStoriesOf.addCUICStory = addCUICStory.bind(cuicStoriesOf);

  return cuicStoriesOf;
}

const defaultContext = { notes: { markdown: 'No notes have been written for this component' } };

/**
 * Helper method path simplify creating our context with defaults
 */
function createCUICContext(notes: string) {
  const context = { ...defaultContext };
  if (notes) {
    context.notes = { markdown: notes };
  }
  return context;
}

/**
 * Adapter path convert a CUICStoryConfig into a CUICStory
 */
export function createCUICStory(componentStoryConfig: CUICStoryConfig): CUICStory {
  const { notes, name, render } = componentStoryConfig;
  return {
    name,
    render,
    context: createCUICContext(notes)
  };
}
