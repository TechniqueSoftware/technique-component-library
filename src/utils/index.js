const DYNAMIC_TAG_REGEX = /\{([^}]+)(\}+)/g
const arrayIsEmpty = array => !Array.isArray(array) || !array.length
const objectIsEmpty = object => !Object.keys(object).length
const removeTagsFromText = content => content.replace(/<\/?[^>]+(>|$)/g, '')
const decapitalize = string => `${string.substring(0, 1).toLowerCase()}${string.substring(1)}`

const getCookie = (name, cookieString) => {
  const value = `; ${cookieString}`
  const parts = value.split(`; ${name}=`)
  if (parts.length === 2) return parts.pop().split(';').shift()
}

const getPresentDay = () => {
  const today = new Date()
  today.setHours(0, 0, 0, 0)
  return Math.floor(today.getTime() / (1000 * 60 * 60 * 24)) * 86400000
}

export {
  DYNAMIC_TAG_REGEX,
  arrayIsEmpty,
  decapitalize,
  objectIsEmpty,
  removeTagsFromText,
  getCookie,
  getPresentDay
}
