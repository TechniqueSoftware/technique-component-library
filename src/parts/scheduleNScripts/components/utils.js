import { STATUSES_WITH_SEGMENTS } from '@parts/scheduleNScripts/constants'
// SEGMENTS
export const checkIfAttributeIsInUse = (segments, selectedAttributes, valueToCheck, segmentId) => {
  if (valueToCheck === '') return false
  const listToCheck = segments.reduce(
    (acc, segment) => {
      if (segmentId && segment.segmentId === segmentId) return acc
      return acc.concat(segment.segmentAttributes)
    },
    selectedAttributes.map(({ value }) => ({ attributeValue: value }))
  )
  return !!listToCheck.find(({ attributeValue }) => attributeValue === valueToCheck)
}

export const getSegmentStatusProp = ({ followUpType, followUpStatus }, keyword) => {
  const segmentTp = STATUSES_WITH_SEGMENTS.find(
    s => s.type === parseInt(followUpType, 10) && s.id === parseInt(followUpStatus, 10)
  )
  return segmentTp?.[keyword]
}

export const parseAttributes = (attributes, type) => attributes.reduce((acc, attr) => {
  if (!attr.value) return acc
  return acc.concat({ attributeValue: attr.value, attributeType: type, attributeId: attr.id })
}, [])

export const unparseAttributes = attributes => attributes.map(({ attributeValue, attributeId }) => ({
  name: attributeValue,
  value: attributeValue,
  id: attributeId
}))
