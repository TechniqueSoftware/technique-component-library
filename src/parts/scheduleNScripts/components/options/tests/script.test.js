import React from 'react'
import * as redux from 'react-redux'
import { render, screen } from '@testing-library/react'
import { TestProvider as Provider } from '@redux/utils'
import { data, locationState, dynamicTagState } from './mock'

import Script from '../script'

const props = {
  dataSchedule: {
    followUpTime: '10:00:00',
    primaryAction: 2,
    secondaryAction: 3,
    followUpMinutes: ''
  },
  setValidMinute: jest.fn(),
  followUpToggle: 'auto'
}

const mockDispatch = jest.fn()
jest.spyOn(redux, 'useDispatch').mockReturnValue(mockDispatch)
jest.spyOn(redux, 'useSelector').mockImplementation(cb => cb({
  global: {
    locations: locationState,
    dynamicTags: dynamicTagState
  },
  scheduleNScripts: { touchpointCrud: { data, action: 1 } }
}))

describe('<Script />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider>
        <Script {...props} />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders tabs', () => {
    render(
      <Provider>
        <Script {...props} />
      </Provider>
    )
    expect(screen.queryByText('Call')).toBeInTheDocument()
    expect(screen.queryByText('Email')).toBeInTheDocument()
    expect(screen.queryByText('Text')).toBeInTheDocument()
  })

  it('render dynamic', () => {
    render(
      <Provider>
        <Script {...props} />
      </Provider>
    )
    expect(screen.getAllByText('Add Dynamic Content')).toHaveLength(3)
  })
})
