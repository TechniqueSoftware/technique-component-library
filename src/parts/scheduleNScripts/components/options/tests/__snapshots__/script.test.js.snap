// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Script /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="tabs tabUnderline nav nav-tabs"
      role="tablist"
    >
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="sns-script-modal-tabs-tabpane-0"
          aria-selected="true"
          class="nav-link active"
          data-rr-ui-event-key="0"
          id="sns-script-modal-tabs-tab-0"
          role="tab"
          type="button"
        >
          <div
            class="
d-flex
justify-content-center
align-items-center
j
"
          >
            <div
              aria-label="BsTelephone"
              class="
    d-flex
    justify-content-center
    align-items-center
     fs-5
    
  "
              role="img"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.654 1.328a.678.678 0 0 0-1.015-.063L1.605 2.3c-.483.484-.661 1.169-.45 1.77a17.568 17.568 0 0 0 4.168 6.608 17.569 17.569 0 0 0 6.608 4.168c.601.211 1.286.033 1.77-.45l1.034-1.034a.678.678 0 0 0-.063-1.015l-2.307-1.794a.678.678 0 0 0-.58-.122l-2.19.547a1.745 1.745 0 0 1-1.657-.459L5.482 8.062a1.745 1.745 0 0 1-.46-1.657l.548-2.19a.678.678 0 0 0-.122-.58L3.654 1.328zM1.884.511a1.745 1.745 0 0 1 2.612.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
                />
              </svg>
            </div>
            <span
              style="margin-left: 8px;"
            >
               Call
            </span>
          </div>
        </button>
      </li>
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="sns-script-modal-tabs-tabpane-1"
          aria-selected="false"
          class="nav-link"
          data-rr-ui-event-key="1"
          id="sns-script-modal-tabs-tab-1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <div
            class="
d-flex
justify-content-center
align-items-center
j
"
          >
            <div
              aria-label="BsEnvelope"
              class="
    d-flex
    justify-content-center
    align-items-center
     fs-5
    
  "
              role="img"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V4zm2-1a1 1 0 0 0-1 1v.217l7 4.2 7-4.2V4a1 1 0 0 0-1-1H2zm13 2.383-4.758 2.855L15 11.114v-5.73zm-.034 6.878L9.271 8.82 8 9.583 6.728 8.82l-5.694 3.44A1 1 0 0 0 2 13h12a1 1 0 0 0 .966-.739zM1 11.114l4.758-2.876L1 5.383v5.73z"
                />
              </svg>
            </div>
            <span
              style="margin-left: 8px;"
            >
               Email
            </span>
          </div>
        </button>
      </li>
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="sns-script-modal-tabs-tabpane-2"
          aria-selected="false"
          class="nav-link"
          data-rr-ui-event-key="2"
          id="sns-script-modal-tabs-tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <div
            class="
d-flex
justify-content-center
align-items-center
j
"
          >
            <div
              aria-label="BsChatDots"
              class="
    d-flex
    justify-content-center
    align-items-center
     fs-5
    
  "
              role="img"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm4 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm3 1a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
                />
                <path
                  d="m2.165 15.803.02-.004c1.83-.363 2.948-.842 3.468-1.105A9.06 9.06 0 0 0 8 15c4.418 0 8-3.134 8-7s-3.582-7-8-7-8 3.134-8 7c0 1.76.743 3.37 1.97 4.6a10.437 10.437 0 0 1-.524 2.318l-.003.011a10.722 10.722 0 0 1-.244.637c-.079.186.074.394.273.362a21.673 21.673 0 0 0 .693-.125zm.8-3.108a1 1 0 0 0-.287-.801C1.618 10.83 1 9.468 1 8c0-3.192 3.004-6 7-6s7 2.808 7 6c0 3.193-3.004 6-7 6a8.06 8.06 0 0 1-2.088-.272 1 1 0 0 0-.711.074c-.387.196-1.24.57-2.634.893a10.97 10.97 0 0 0 .398-2z"
                />
              </svg>
            </div>
            <span
              style="margin-left: 8px;"
            >
               Text
            </span>
          </div>
        </button>
      </li>
    </ul>
    <div
      class="tab-content"
    >
      <div
        aria-labelledby="sns-script-modal-tabs-tab-0"
        class="fade border border-top-0 rounded-bottom shadow-sm border-0 p-3  tab-pane active show"
        id="sns-script-modal-tabs-tabpane-0"
        role="tabpanel"
      >
        <div
          class="row"
        >
          <div
            class="form-col col"
          >
            <h5>
              Follow-Up Script: Member Day 166
            </h5>
            <form
              class=""
            >
              <div
                class="row"
              >
                <div
                  class="mb-3 col-sm-8"
                >
                  <div>
                    <div
                      class="row"
                    >
                      <label
                        class="form-label col-form-label col-form-label-lg col-lg-5"
                        for="dynamicContent1"
                      >
                        Add Dynamic Content
                      </label>
                      <div
                        class="col"
                      >
                        <select
                          aria-label="Select Add Dynamic Content"
                          class="form-select form-select-md"
                          disabled=""
                          id="dynamicContent1"
                          required=""
                          role="combobox"
                          type="select"
                        >
                          <optgroup
                            label="Default"
                          >
                            <option
                              value=""
                            >
                              Select to Add
                            </option>
                          </optgroup>
                          <optgroup
                            label="Recipient"
                          >
                            <option
                              value="{{recipient-first}}"
                            >
                              First Name
                            </option>
                          </optgroup>
                        </select>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="col-sm-4"
                />
              </div>
              <div
                class="row"
                id="editortext"
              >
                <div
                  class="readonlyScript mx-auto mb-2 p-2"
                >
                  <p>
                    This is test script
                  </p>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="sns-script-modal-tabs-tab-1"
        class="fade border border-top-0 rounded-bottom shadow-sm border-0 p-3  tab-pane"
        id="sns-script-modal-tabs-tabpane-1"
        role="tabpanel"
      >
        <div
          class="row"
        >
          <div
            class="form-col col"
          >
            <h5>
              Follow-Up Script: Member Day 166
            </h5>
            <form
              class=""
            >
              <div
                class="row"
              >
                <div
                  class="mb-3 col-sm-8"
                >
                  <div>
                    <div
                      class="row"
                    >
                      <label
                        class="form-label col-form-label col-form-label-lg col-lg-5"
                        for="dynamicContent2"
                      >
                        Add Dynamic Content
                      </label>
                      <div
                        class="col"
                      >
                        <select
                          aria-label="Select Add Dynamic Content"
                          class="form-select form-select-md"
                          disabled=""
                          id="dynamicContent2"
                          required=""
                          role="combobox"
                          type="select"
                        >
                          <optgroup
                            label="Default"
                          >
                            <option
                              value=""
                            >
                              Select to Add
                            </option>
                          </optgroup>
                          <optgroup
                            label="Recipient"
                          >
                            <option
                              value="{{recipient-first}}"
                            >
                              First Name
                            </option>
                          </optgroup>
                        </select>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="col-sm-4"
                />
              </div>
              <div
                class="row"
                id="editortext"
              >
                <div
                  class="readonlyScript mx-auto mb-2 p-2"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="sns-script-modal-tabs-tab-2"
        class="fade border border-top-0 rounded-bottom shadow-sm border-0 p-3  tab-pane"
        id="sns-script-modal-tabs-tabpane-2"
        role="tabpanel"
      >
        <div
          class="row"
        >
          <div
            class="form-col col"
          >
            <h5>
              Follow-Up Script: Member Day 166
            </h5>
            <form
              class=""
            >
              <div
                class="row"
              >
                <div
                  class="mb-3 col-sm-8"
                >
                  <div>
                    <div
                      class="row"
                    >
                      <label
                        class="form-label col-form-label col-form-label-lg col-lg-5"
                        for="dynamicContent3"
                      >
                        Add Dynamic Content
                      </label>
                      <div
                        class="col"
                      >
                        <select
                          aria-label="Select Add Dynamic Content"
                          class="form-select form-select-md"
                          disabled=""
                          id="dynamicContent3"
                          required=""
                          role="combobox"
                          type="select"
                        >
                          <optgroup
                            label="Default"
                          >
                            <option
                              value=""
                            >
                              Select to Add
                            </option>
                          </optgroup>
                          <optgroup
                            label="Recipient"
                          >
                            <option
                              value="{{recipient-first}}"
                            >
                              First Name
                            </option>
                          </optgroup>
                        </select>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="col-sm-4"
                />
              </div>
              <div
                class="row"
                id="editortext"
              >
                <div
                  class="readonlyScript mx-auto mb-2 p-2"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
